### PR TITLE
Reactive datasource config: cache-prepared-statements is not specific to any DB

### LINKED ISSUE #9642: Reactive datasource config: cache-prepared-statements is not specific to any DB
`cache-prepared-statements` config attribute is duplicated in MySQL and Postgres configs.

It should be a common attribute of any Reactive SQL client.

### PR DESCRIPTION
**Pull request by tsegismont**
Fixes #9642

### CODE REVIEW

**Comment by tsegismont** [comment]
Created a draft PR for now as it is based on top of #9469 .

The only meaningful commit is a923557

**Comment by tsegismont** [comment]
@cescoffier @FroMage can one of you please review? Thank you

File: extensions/reactive-mysql-client/runtime/src/main/java/io/quarkus/reactive/mysql/client/runtime/MySQLPoolRecorder.java [hunk 1]
```diff
@@ -119,8 +124,12 @@ private MySQLConnectOptions toMySQLConnectOptions(DataSourceRuntimeConfig dataSo
         }
 
         if (dataSourceReactiveMySQLConfig.cachePreparedStatements.isPresent()) {
+            log.warn("cache-prepared-statements is not specific to a database kind, configure it at the parent level");
```
Comment (FroMage) [inline extensions/reactive-mysql-client/runtime/src/main/java/io/quarkus/reactive/mysql/client/runtime/MySQLPoolRecorder.java:127]: If you include the key name in the message it's even more helpful.

File: extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/runtime/PgPoolRecorder.java [hunk 2]
```diff
@@ -121,8 +125,12 @@ private PgConnectOptions toPgConnectOptions(DataSourceRuntimeConfig dataSourceRu
         }
 
         if (dataSourceReactivePostgreSQLConfig.cachePreparedStatements.isPresent()) {
+            log.warn("cache-prepared-statements is not specific to a database kind, configure it at the parent level");
```
Comment (FroMage) [inline extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/runtime/PgPoolRecorder.java:128]: Same remark.

**Review by FroMage — CHANGES_REQUESTED** [review]
Except for a better warning, this is good to merge.

File: extensions/reactive-mysql-client/runtime/src/main/java/io/quarkus/reactive/mysql/client/runtime/MySQLPoolRecorder.java (hunk 1 above)
Comment (tsegismont) [reply extensions/reactive-mysql-client/runtime/src/main/java/io/quarkus/reactive/mysql/client/runtime/MySQLPoolRecorder.java]: Done!

File: extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/runtime/PgPoolRecorder.java (hunk 2 above)
Comment (tsegismont) [reply extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/runtime/PgPoolRecorder.java]: Done!

**Review by FroMage — APPROVED** [review]

**Comment by Sanne** [comment]
great, thanks!

**Comment by tsegismont** [comment]
Thank you @Sanne and @FroMage !

**Pull request merged**